src/AspNetCore/Src/applicationinfo.cpp

```diff
@@ -23,8 +23,7 @@ APPLICATION_INFO::~APPLICATION_INFO()
 
     if (m_pApplication != NULL)
     {
-        // shutdown the application
-        m_pApplication->ShutDown();
+        // Shutdown should have already been called if gracefully shutting down.
```

Comment:
if you believe shutdown has already been done, maybe you need a flag to address it. Otherwise, if the shutdown is in the middle with another thread and this thread deference it and delete it, some AV may happen.

Reply:
I like adding a flag for knowing if we already called shutdown in the application info. The idea behind this PR is I wanted to separate Shutdown from the destructor entirely, because shutdown can be triggered by either native or managed. I can discuss with you tomorrow about this.